runtime/compiler/env/j9method.cpp

```diff
@@ -2387,15 +2393,48 @@ TR_ResolvedJ9Method::TR_ResolvedJ9Method(TR_FrontEnd * fe, TR_ResolvedMethod * o
 
 void TR_ResolvedJ9Method::construct()
    {
-#define x(a, b, c) a, sizeof(b) - 1, b, (int16_t)strlen(c), c
-
    struct X
       {
       TR::RecognizedMethod _enum;
       char _nameLen;
       const char * _name;
       int16_t _sigLen;
       const char * _sig;
+
+      typedef int8_t javaVersion_t;
+
+      javaVersion_t _nativeMinVersion;
+      javaVersion_t _nativeMaxVersion;
+
+// AllPastJavaVer and AllFutureJavaVer are used as lower and upper bounds for
+// ranges of Java versions over which a recognized method is expected to be
+// native, where that range is (currently) unbounded.  An upper bound of
+// AllPastJaveVer is used if the method has never been expected to be native.
+#define AllPastJavaVer   INT8_MIN
+#define AllFutureJavaVer INT8_MAX
+
+// xClMeth, xSig, xWCSig and xNonNativeRange are intermediate macros used in
+// the macros used to initialize entries in the tables of recognized methods
+#define xClMeth(cl, meth) cl, sizeof(meth) - 1, meth
+#define xSig(sig)         (int16_t) sizeof(sig) - 1, sig
+#define xWCSig            (int16_t) -1, "*"
+#define xNonNativeRange   AllPastJavaVer, AllPastJavaVer
+
+// The following four macros are used to intialize entries in the tables
+// of recognized methods:
+//
+//   - x is used for methods that with specific signatures that are not expected
+//     to be native
+//   - xAnySig is used for methods with a specific name, but arbitrary
+//     signature, that are not expected to be native
+//   - xNative is used for methods that with specific signatures that are
+//     expected to be native
+//   - xAnySigNative is used for methods with a specific name, but arbitrary/
+//     signature, that are expected to be native
+#define x(cl, meth, sig) xClMeth(cl, meth), xSig(sig), xNonNativeRange
```

Comment:
As above, could you spell out the parameters to aid others reading the code? `lb` and `ub` are a bit cryptic - something like minVersion and maxVersion or something like that. A comment on the bounds being inclusive probably would be helpful. Eg that a min max of 9, 10 would be in versions 9 and 10 so that there is no confusion.